testing_dir/minesweeper.py

```diff
@@ -89,7 +89,7 @@ def game_loop():
             player_pos = new_pos
             moves += 1
 
-        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
+        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES + 1:  # Error: should not have +1
```

Comment:
The condition in the if statement seems incorrect. The "+1" should not be there as it might cause the game to end prematurely.
```diff
-        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES + 1:
+        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
```